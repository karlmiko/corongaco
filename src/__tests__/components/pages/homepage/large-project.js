import React from 'react'
import renderer from 'react-test-renderer'
import { useStaticQuery } from 'gatsby'
import LargeProject from '~components/pages/homepage/large-project'

beforeEach(() => {
  useStaticQuery.mockImplementation(() => ({
    file: {
      relativePath: 'project-large.png',
      childImageSharp: {
        fluid: {
          aspectRatio: 1.3,
          sizes: '(max-width: 1200px) 100vw, 1200px',
          src:
            '/static/d7b5369bedd2904efd15de13fa55daef/f3583/project-large.png',
          srcSet: '/srcset',
          tracedSVG: '[SVG]',
        },
      },
    },
  }))
})

describe('Components : Pages : Homepage : Large Project', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<LargeProject />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
