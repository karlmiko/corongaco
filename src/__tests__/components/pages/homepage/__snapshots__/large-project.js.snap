// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components : Pages : Homepage : Large Project renders correctly 1`] = `
<div
  className="wrapper"
>
  <div
    className="container"
  >
    <a
      href="/race"
    >
      <div
        className=" gatsby-image-wrapper"
        style={
          Object {
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          aria-hidden={true}
          style={
            Object {
              "paddingBottom": "76.92307692307692%",
              "width": "100%",
            }
          }
        />
        <img
          alt=""
          aria-hidden={true}
          src="[SVG]"
          style={
            Object {
              "height": "100%",
              "left": 0,
              "objectFit": "cover",
              "objectPosition": "center",
              "opacity": 1,
              "position": "absolute",
              "top": 0,
              "transitionDelay": "500ms",
              "width": "100%",
            }
          }
        />
        <picture>
          <source
            sizes="(max-width: 1200px) 100vw, 1200px"
            srcSet="/srcset"
          />
          <img
            alt="Illustration of a face split into strips of different colored skintones on a light purple background."
            loading="lazy"
            onLoad={[Function]}
            sizes="(max-width: 1200px) 100vw, 1200px"
            src="/static/d7b5369bedd2904efd15de13fa55daef/f3583/project-large.png"
            srcSet="/srcset"
            style={
              Object {
                "height": "100%",
                "left": 0,
                "objectFit": "cover",
                "objectPosition": "center",
                "opacity": 0,
                "position": "absolute",
                "top": 0,
                "transition": "opacity 500ms",
                "width": "100%",
              }
            }
          />
        </picture>
        <noscript
          dangerouslySetInnerHTML={
            Object {
              "__html": "<picture><source srcset=\\"/srcset\\" sizes=\\"(max-width: 1200px) 100vw, 1200px\\" /><img loading=\\"lazy\\" sizes=\\"(max-width: 1200px) 100vw, 1200px\\" srcset=\\"/srcset\\" src=\\"/static/d7b5369bedd2904efd15de13fa55daef/f3583/project-large.png\\" alt=\\"Illustration of a face split into strips of different colored skintones on a light purple background.\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
            }
          }
        />
      </div>
    </a>
    <div
      className="credit"
    >
      Mohini Dutta / COVID Tracking Project
    </div>
    <h3>
      <a
        href="/race"
      >
        COVID-19 isn't affecting all communities equally. We've partnered with the Antiracist Research & Policy Center to collect the most complete racial data anywhere in the COVID Racial Data Tracker.
      </a>
    </h3>
    <p>
      We're tracking racial and ethnic data from every state that reports it—and pushing those that don't to start. Together with the Antiracist Research & Policy Center, we're analyzing this data to uncover the true impact of the outbreak on vulnerable communities.
    </p>
    <a
      className="cta centered "
      href="/race"
    >
      See the racial data tracker
      <span
        aria-hidden={true}
        className="arrow"
      >
         
        →
      </span>
    </a>
  </div>
</div>
`;
